tests/cpp_validation_tests/CMakeLists.txt

```diff
@@ -12,6 +12,8 @@ add_executable(power_grid_model_validation_tests ${PROJECT_SOURCES})
 target_link_libraries(power_grid_model_validation_tests
     PRIVATE
         power_grid_model
```

Comment:
Remove this one
Add a dependency for `nolhmann_json`

Reply:
I will after I have removed all the dependencies on the core. Leaving it for now. I will resolve this comment once it is fully removed.

Reply:
All the dependencies have been removed. Resolving.